06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js

```diff
@@ -69,90 +69,100 @@ const InvertedIndex = class {
    * @return {String} sanitizedText - sanitized contents of each document
    */
   static tokenize(text) {
-    let sanitizedText = text.replace(/[^\w\s]+/gi, '');
-    sanitizedText = sanitizedText.replace(/\s\s+/g, ' ');
-    sanitizedText = sanitizedText.replace(/^[.\s]+|[.\s]+$/g, '');
-    return sanitizedText.toLowerCase();
-  }
-
-  setBookTitles(bookname, title) {
-    if (this.indexedBookTitles[bookname]) {
-      this.indexedBookTitles[bookname].push(title);
-    } else {
-      this.indexedBookTitles[bookname] = [];
-      this.indexedBookTitles[bookname].push(title);
-    }
+    return text.replace(/[^\w\s]/gi, '')
+      .replace(/\s+/g, ' ')
+      .trim()
+      .toLowerCase();
   }
 
   /**
    * Creates a word array for each document in a book
+   * @param {String} bookName - name of a book to index
    * @param {Object} book - documents in a book
    * @return {Array.<Object>} bookContents - words in each document
    */
-  createsArray(bookname, book) {
+  createsArray(bookName, book) {
     const bookContents = [];
     Object.keys(book).map((documentPosition) => {
       const mergedTitleAndText = `${book[documentPosition].title} 
       ${book[documentPosition].text}`;
       bookContents.push(InvertedIndex.tokenize(mergedTitleAndText).split(' '));
-      this.setBookTitles(bookname, book[documentPosition].title);
+      this.setBookTitles(bookName, book[documentPosition].title);
+      return this;
```

Comment:
From study, by default functional methods such as map return an implicit 'this'. Which is used to support method chaining.
In this scenario, I am returning this tokenIndex[eachWord], thereby making it possible to chain other methods on the map method. I think what this means is, whether I explicitly do it or not, it is done internally. But I have done it to avoid Eslint flags.

I have however, changed it to explicitly return tokenIndex[eachWord]